packages/native/src/client.rs

```diff
@@ -70,34 +79,46 @@ impl FFIClient {
         }
     }
 
-    pub fn get_env_by_uri(&self, uri: Arc<Uri>) -> Option<String> {
-        match self.inner_client.get_env_by_uri(&uri) {
-            Some(env) => Some(env.to_string()),
-            None => None,
-        }
+    pub fn get_env_by_uri(&self, uri: Arc<FFIUri>) -> Option<Vec<u8>> {
+        self.inner_client.get_env_by_uri(&uri.0).map(|e| e.to_vec())
     }
 
     pub fn invoke_wrapper_raw(
         &self,
-        wrapper: Arc<FFIWrapper>,
-        uri: Arc<Uri>,
+        wrapper: Box<dyn FFIWrapper>,
+        uri: Arc<FFIUri>,
```

Comment:
We could treat Uri parameters as `&str` or `String` (no need for `Arc` as `String` nor `&str` are interfaces) but I'd rather have a single implementation of `Uri` to be used/consumed; and also `FFIUri` is a stronger constraint than `String` as it would force the consumer to validly instantiate a URI from `String` before invoking. But those are just my thoughts